### PR TITLE
rosdep: add python-igraph to rosdep pip

### PR DESCRIPTION
**Pull request by duaneenorris**
Python interface to the igraph high performance graph library,
primarily aimed at complex network research and analysis.

Package link:
https://pypi.org/project/python-igraph/

Please add the following dependency to the rosdep database.

## Package name:

python-igraph 

## Package Upstream Source:

https://pypi.org/project/python-igraph/

## Purpose of using this:

Python interface to the igraph high performance graph library,
primarily aimed at complex network research and analysis.

Distro packaging links:

## Links to Distribution Packages

- Debian: https://packages.debian.org/
  - https://pypi.org/project/python-igraph/

- Ubuntu: https://packages.ubuntu.com/
   - https://pypi.org/project/python-igraph/

- Fedora: https://src.fedoraproject.org/browse/projects/
  - https://pypi.org/project/python-igraph/

- Arch: https://www.archlinux.org/packages/
  - IF AVAILABLE
- Gentoo: https://packages.gentoo.org/
  - IF AVAILABLE
- macOS: https://formulae.brew.sh/
  - IF AVAILABLE
- Alpine: https://pkgs.alpinelinux.org/packages
  - IF AVAILABLE
- NixOS/nixpkgs: https://search.nixos.org/packages
  - OPTIONAL

# Please Add This Package to be indexed in the rosdistro.

ROSDISTRO NAME

# The source is here: 

https://pypi.org/project/python-igraph/

# Checks
 - [ ] All packages have a declared license in the package.xml
 - [ ] This repository has a LICENSE file
 - [ ] This package is expected to build on the submitted rosdistro

### CODE REVIEW

**Review by ivanpauno — APPROVED** [review]

File: rosdep/python.yaml [hunk 1]
```diff
@@ -2216,6 +2216,37 @@ python-hypothesis:
...
+      packages: [python-igraph]
+  debian:
+    pip:
+      packages: [python-igraph]
+  fedora:
+    pip:
+      packages: [python-igraph]
+  gentoo:
+    pip:
+      packages: [python-igraph]
```
Comment (sloretz) [inline rosdep/python.yaml:2234]: Pip can't be used on Gentoo according to the CONTRIBUTING guide. Mind deleting this?

https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md#gentoo

Comment (duaneenorris) [reply rosdep/python.yaml]: Removed

**Review by tfoote — CHANGES_REQUESTED** [review]
I just checked quickly and there are native packages for this which should be used instead of the pip version: https://packages.ubuntu.com/bionic/python3-igraph

Please update this to use the native packages for the platforms outline in the contributing guide.

**Comment by duaneenorris** [comment]
Since this package is available as a native package, closing PR.

**Pull request closed without merging**